### PR TITLE
Migrate the rest of Libs Package.

### PR DESCRIPTION
**Pull request by tneymanov**

### CODE REVIEW

File: gcp_variant_transforms/beam_io/vcf_estimate_io.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 """A source for reading VCF files and extracting signals about input size."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_estimate_io.py:18]: line.

File: gcp_variant_transforms/beam_io/vcf_header_io.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 """A source for reading VCF file headers."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_header_io.py:17]: line.

File: gcp_variant_transforms/beam_io/vcf_header_io.py [hunk 3]
```diff
@@ -589,9 +589,9 @@ def _format_number(self, number):
       raise ValueError('Invalid value for number: {}'.format(number))
 
   def _format_string_value(self, value):
-    # type: (str, unicode) -> str
-    if isinstance(value, unicode):
-      return '"{}"'.format(value.encode('utf-8'))
+    # type: (str, bytes) -> str
+    if isinstance(value, bytes):
+      return '"{}"'.format(value.decode('utf-8'))
```
Comment (samanvp) [inline gcp_variant_transforms/beam_io/vcf_header_io.py:594]: I just noticed this PR is not synced with #646

**Review by samanvp — COMMENTED** [review]
Could you please sync with #646 to avoid redundant comments

File: gcp_variant_transforms/libs/partitioning.py [hunk 4]
```diff
@@ -115,7 +114,7 @@ def _run_query(self, query):
         break
     result = []
     for i in iterator:
-      result.append(str(i.values()[0]))
+      result.append(str(list(i.values())[0]))
```
Comment (samanvp) [inline gcp_variant_transforms/libs/partitioning.py:117]: I am not sure if this list is needed?

File: gcp_variant_transforms/libs/partitioning_test.py [hunk 5]
```diff
@@ -30,7 +30,7 @@ def test_calculate_optimal_range_interval(self):
         40030000: 10020,
     }
     for range_end, expected_range_interval in (
-        range_end_to_expected_range_interval.items()):
+        list(range_end_to_expected_range_interval.items())):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/partitioning_test.py:33]: Remove list.

File: gcp_variant_transforms/libs/preprocess_reporter.py [hunk 6]
```diff
@@ -115,7 +115,7 @@ def _extract_conflicts(
     which maps `Definition` to a list of file names.
   """
   # len(v) > 1 means there are conflicting definitions for this field.
-  return dict([(k, v) for k, v in definitions.items() if len(v) > 1])
+  return {k:v for k, v in list(definitions.items()) if len(v) > 1}
```
Comment (samanvp) [inline gcp_variant_transforms/libs/preprocess_reporter.py:118]: Remove list.

File: gcp_variant_transforms/libs/processed_variant.py [hunk 7]
```diff
@@ -20,7 +20,7 @@
 functions are "private".
 """
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant.py:23]: line.

File: gcp_variant_transforms/libs/processed_variant.py [hunk 8]
```diff
@@ -255,7 +255,7 @@ def create_processed_variant(self, variant):
     """
     proc_var = ProcessedVariant(variant)
     self._variant_counter.inc()
-    for key, variant_info_data in variant.info.iteritems():
+    for key, variant_info_data in list(variant.info.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant.py:258]: Remove list.

File: gcp_variant_transforms/libs/processed_variant.py [hunk 9]
```diff
@@ -302,7 +302,7 @@ def create_alt_bases_field_schema(self):
         mode=bigquery_util.TableFieldConstants.MODE_NULLABLE,
         description='Alternate base.'))
     if self._split_alternate_allele_info_fields:
-      for key, field in self._header_fields.infos.iteritems():
+      for key, field in list(self._header_fields.infos.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant.py:305]: Remove list.

File: gcp_variant_transforms/libs/processed_variant.py [hunk 10]
```diff
@@ -475,7 +475,7 @@ def add_annotation_data(self, proc_var, annotation_field_name, data):
     for annotation_str in data:
       try:
         ind, annotation_map = parser.parse_and_match_alt(annotation_str)
-        for name, value in annotation_map.iteritems():
+        for name, value in list(annotation_map.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant.py:478]: Remove list.

File: gcp_variant_transforms/libs/processed_variant_test.py [hunk 11]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant_test.py:15]: line.

File: gcp_variant_transforms/libs/processed_variant_test.py [hunk 12]
```diff
@@ -534,7 +534,7 @@ def test_create_processed_variant_annotation_alt_allele_num(self):
 
   def test_create_alt_bases_field_schema_descriptions(self):
     _, header_fields = self._get_sample_variant_and_header_with_csq()
-    for hfi in header_fields.infos.values():
+    for hfi in list(header_fields.infos.values()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant_test.py:537]: Remove list.

File: gcp_variant_transforms/libs/processed_variant_test.py [hunk 13]
```diff
@@ -563,7 +563,7 @@ def test_create_alt_bases_field_schema_types(self):
              for i, t in zip(ids, types)]
     _, header_fields = self._get_sample_variant_and_header_with_csq(
         additional_infos=infos)
-    for hfi in header_fields.infos.values():
+    for hfi in list(header_fields.infos.values()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/processed_variant_test.py:566]: Remove list.

File: gcp_variant_transforms/libs/schema_converter.py [hunk 14]
```diff
@@ -14,7 +14,7 @@
 
 """Handles the conversion between BigQuery/Avro schema and VCF header."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/schema_converter.py:17]: line.

File: gcp_variant_transforms/libs/schema_converter.py [hunk 15]
```diff
@@ -147,7 +147,7 @@ def generate_schema_from_header_fields(
       description=('Phaseset of the call (if any). "*" is used in cases where '
                    'the genotype is phased, but no phase set ("PS" in FORMAT) '
                    'was specified.')))
-  for key, field in header_fields.formats.iteritems():
+  for key, field in list(header_fields.formats.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/schema_converter.py:150]: Remove list.

File: gcp_variant_transforms/libs/schema_converter.py [hunk 16]
```diff
@@ -165,7 +165,7 @@ def generate_schema_from_header_fields(
   info_keys = set()
   annotation_info_type_keys_set = set(
       proc_variant_factory.gen_annotation_info_type_keys())
-  for key, field in header_fields.infos.iteritems():
+  for key, field in list(header_fields.infos.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/schema_converter.py:168]: Remove list.

File: gcp_variant_transforms/libs/schema_converter_test.py [hunk 17]
```diff
@@ -14,7 +14,7 @@
 
 """Tests for `schema_converter` module."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/schema_converter_test.py:17]: line.

File: gcp_variant_transforms/libs/schema_converter_test.py [hunk 18]
```diff
@@ -218,7 +218,7 @@ class ConvertTableSchemaToJsonAvroSchemaTest(
   """
 
   def _validate_schema(self, expected_fields, actual_schema):
-    super(ConvertTableSchemaToJsonAvroSchemaTest, self)._validate_schema(
+    super()._validate_schema(
         expected_fields, actual_schema)
```
Comment (samanvp) [inline gcp_variant_transforms/libs/schema_converter_test.py:222]: nit: Can be single line.

File: gcp_variant_transforms/libs/variant_sharding.py [hunk 19]
```diff
@@ -24,7 +24,7 @@
      available at gcp_variant_transforms/testing/data/misc/*.yaml
 """
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/variant_sharding.py:27]: line.

File: gcp_variant_transforms/libs/variant_sharding_test.py [hunk 20]
```diff
@@ -14,7 +14,7 @@
 
 """Unit tests for variant_sharding module."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/variant_sharding_test.py:17]: line.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py [hunk 21]
```diff
@@ -41,12 +41,12 @@ def __init__(self, vcf_header=None):
     self._formats = collections.defaultdict(dict)
     if not vcf_header:
       return
-    for key, val in vcf_header.infos.iteritems():
+    for key, val in list(vcf_header.infos.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_header_definitions_merger.py:44]: Remove list.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py [hunk 22]
```diff
@@ -41,12 +41,12 @@ def __init__(self, vcf_header=None):
...
       definition = Definition(
           val[vcf_header_io.VcfParserHeaderKeyConstants.NUM],
           val[vcf_header_io.VcfParserHeaderKeyConstants.TYPE])
       self._infos[key][definition] = [vcf_header.file_path]
-    for key, val in vcf_header.formats.iteritems():
+    for key, val in list(vcf_header.formats.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_header_definitions_merger.py:49]: Remove list.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py [hunk 23]
```diff
@@ -86,8 +86,8 @@ def _merge_definitions(
       ):
     # type: (...) -> None
     """Updates `first` by merging values from `first` and `second`."""
-    for key, definitions_to_files_map in second.iteritems():
-      for definition, file_names in definitions_to_files_map.iteritems():
+    for key, definitions_to_files_map in list(second.items()):
+      for definition, file_names in list(definitions_to_files_map.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_header_definitions_merger.py:90]: 2 * Remove list.

File: gcp_variant_transforms/libs/vcf_header_parser.py [hunk 24]
```diff
@@ -14,7 +14,7 @@
 
 """Helper library for reading VCF headers from multiple files."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_header_parser.py:17]: line

File: gcp_variant_transforms/libs/vcf_header_parser_test.py [hunk 25]
```diff
@@ -14,7 +14,7 @@
 
 """Tests for vcf_header_parser module."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_header_parser_test.py:17]: line.

File: gcp_variant_transforms/libs/vcf_reserved_fields.py [hunk 26]
```diff
@@ -17,7 +17,7 @@
 See http://samtools.github.io/hts-specs/VCFv4.3.pdf for more details.
 """
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/libs/vcf_reserved_fields.py:20]: line.

File: gcp_variant_transforms/testing/asserts.py [hunk 27]
```diff
@@ -14,7 +14,7 @@
 
 """Custom asserts for tests."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/testing/asserts.py:17]: line

File: gcp_variant_transforms/testing/vcf_header_util.py [hunk 28]
```diff
@@ -14,7 +14,7 @@
 
 """Utility functions for creating VcfHeader objects used by unit tests."""
 
-from __future__ import absolute_import
+
```
Comment (samanvp) [inline gcp_variant_transforms/testing/vcf_header_util.py:17]: line.

File: gcp_variant_transforms/testing/vcf_header_util.py [hunk 29]
```diff
@@ -27,7 +27,7 @@ def make_header(header_num_dict):
     header_num_dict: a dictionary mapping info keys to string num values.
   """
   infos = {}
-  for k, v in header_num_dict.iteritems():
+  for k, v in list(header_num_dict.items()):
```
Comment (samanvp) [inline gcp_variant_transforms/testing/vcf_header_util.py:30]: Remove list.

**Review by samanvp — APPROVED** [review]
A bunch of nits and remove `list()` from `for` loops.

Thanks Tural.

File: gcp_variant_transforms/libs/partitioning_test.py (hunk 5 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/partitioning_test.py]: Done.

File: gcp_variant_transforms/libs/processed_variant.py (hunk 7 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant.py:23]: Done.

File: gcp_variant_transforms/libs/processed_variant.py (hunk 8 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant.py]: Done.

File: gcp_variant_transforms/libs/processed_variant.py (hunk 9 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant.py]: Done.

File: gcp_variant_transforms/libs/processed_variant.py (hunk 10 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant.py]: Done.

File: gcp_variant_transforms/libs/processed_variant_test.py (hunk 11 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant_test.py:15]: No, here line apparently was missing, so needs to stay.

File: gcp_variant_transforms/libs/processed_variant_test.py (hunk 12 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant_test.py]: Done.

File: gcp_variant_transforms/libs/processed_variant_test.py (hunk 13 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/processed_variant_test.py]: Done.

File: gcp_variant_transforms/libs/schema_converter.py (hunk 14 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/schema_converter.py:17]: Done.

File: gcp_variant_transforms/libs/schema_converter.py (hunk 15 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/schema_converter.py]: Done.

File: gcp_variant_transforms/libs/schema_converter.py (hunk 16 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/schema_converter.py]: Done.

File: gcp_variant_transforms/libs/schema_converter_test.py (hunk 17 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/schema_converter_test.py:17]: Done.

File: gcp_variant_transforms/libs/variant_sharding.py (hunk 19 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_sharding.py:27]: Done.

File: gcp_variant_transforms/libs/variant_sharding_test.py (hunk 20 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_sharding_test.py:17]: Done.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py (hunk 21 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_header_definitions_merger.py]: Done.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py (hunk 22 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_header_definitions_merger.py]: Done.

File: gcp_variant_transforms/libs/vcf_header_definitions_merger.py (hunk 23 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_header_definitions_merger.py]: Done.

File: gcp_variant_transforms/libs/vcf_header_parser.py (hunk 24 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_header_parser.py:17]: Done.

File: gcp_variant_transforms/libs/vcf_header_parser_test.py (hunk 25 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_header_parser_test.py:17]: Done.

File: gcp_variant_transforms/libs/vcf_reserved_fields.py (hunk 26 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/vcf_reserved_fields.py:20]: Done.

File: gcp_variant_transforms/testing/asserts.py (hunk 27 above)
Comment (tneymanov) [reply gcp_variant_transforms/testing/asserts.py:17]: Done.

File: gcp_variant_transforms/testing/vcf_header_util.py (hunk 28 above)
Comment (tneymanov) [reply gcp_variant_transforms/testing/vcf_header_util.py:17]: Done.

File: gcp_variant_transforms/testing/vcf_header_util.py (hunk 29 above)
Comment (tneymanov) [reply gcp_variant_transforms/testing/vcf_header_util.py]: Done.

**Pull request merged**